src/exit_status.h

```diff
@@ -15,10 +15,18 @@
 #ifndef NINJA_EXIT_STATUS_H_
 #define NINJA_EXIT_STATUS_H_
 
-enum ExitStatus {
-  ExitSuccess,
+// The underlying type of the ExitStatus enum, used to represent a platform-specific
+// process exit code.
+#ifdef _WIN32
+#define EXIT_STATUS_TYPE unsigned long
+#else  // !_WIN32
+#define EXIT_STATUS_TYPE int
+#endif  // !_WIN32
+
+enum ExitStatus : EXIT_STATUS_TYPE {
+  ExitSuccess=0,
   ExitFailure,
-  ExitInterrupted
+  ExitInterrupted=130
```

Comment:
The value 130 is Posix-specific. On Windows, the convention is that the C runtime will return 3 in case of Ctrl-C, though some programs also return 1. I suggest you define the value with another platform-specific macro, e.g.:

```
// The underlying type of the ExitStatus enum, used to represent a platform-specific
// process exit code.
#ifdef _WIN32
#define EXIT_STATUS_TYPE unsigned long
#else  // !_WIN32
#define EXIT_STATUS_TYPE int
#endif  // !_WIN32

// The platform-specific value of ExitInterrupted returned by Ninja in case of use interruption (e.g. Ctrl-C).
// On Windows, the convention used by the C runtime library is to return 3, even
// though the system error `ERROR_CONTROL_C_EXIT` is 572. Some applications also return 1
// which is ExitFailure. The value 2 is chosen to preserve the existing Ninja behavior on this
// platform. On Posix, this is simply 128 + SIGINT. Note that Ninja will map SIGHUP
// and SIGTERM interrupted to ExitInterrupted as well.
#ifdef _WIN32
#  define EXIT_INTERRUPTED_VALUE  2
#else  // !_WIN32
#  define EXIT_INTERRUPTED_VALUE  130
#endif  // !_WIN32

enum ExitStatus : EXIT_STATUS_TYPE {
  ExitSuccess=0,
  ExitFailure,
  ExitInterrupted = EXIT_INTERRUPTED_VALUED,
};
```

Also, document in the commit message that this changes the value returned by Ninja on Posix for user interruptions. This is important for the future release notes.

Reply:
Thanks. I'll take a look on how to comment the commit properly